Issues passing AiiDA datatype: Dict into tasks in aiida-workgraph v0.7.5

### Issue: 
When passing in `orm.Dict` to task as kwargs, if the workgraph is displayed with the use of `wg` in a notebook the workgraph passes in unwanted args into the task. 
If `wg` is not displayed it runs fine. 

### Steps to reporduce:
On aiida_workgraph v0.7.5:

[wg_test.ipynb](https://github.com/user-attachments/files/23341681/wg_test.ipynb)

Above is a basic example of the issue. 

When we view and run the workgraph we get the following error:

<img width="747" height="1076" alt="Image" src="https://github.com/user-attachments/assets/515836c3-473c-4cff-b4c1-3165c9e8c31e" />

```
11/04/2025 04:50:01 PM <819218> plumpy.processes: [ERROR] Error in task user_info: Invalid assignment into namespace socket: user_fact.inputs

What happened:
  • Field 'node_type' is not defined and this namespace is not dynamic.
  • Incoming value: key 'node_type'

Why this matters:
  • Without a namespace-typed graph input, the whole dict is treated as *one value*,
    so its inner keys are NOT wired to the child sockets. You'd end up with missing links.

How to fix:
  • Add the field to the namespace spec; or
  • make the namespace dynamic if it should grow automatically.
Traceback (most recent call last):
  File "/home/mtr46585/aiida-mlip/.venv/lib/python3.12/site-packages/aiida_workgraph/engine/task_manager.py", line 216, in execute_process_task
    process, state = task.execute(
                     ^^^^^^^^^^^^^
  File "/home/mtr46585/aiida-mlip/.venv/lib/python3.12/site-packages/aiida_workgraph/tasks/graph_task.py", line 55, in execute
    wg = materialize_graph(
         ^^^^^^^^^^^^^^^^^^
  File "/home/mtr46585/aiida-mlip/.venv/lib/python3.12/site-packages/node_graph/utils/graph.py", line 168, in materialize_graph
...
```

However, if we remove `wg` it runs fine:
<img width="1322" height="479" alt="Image" src="https://github.com/user-attachments/assets/545d46cc-cbc1-426e-9f07-30a3c63af7aa" />

### Possible solution:

when we `print(user.value()`  the dictionary has a `node_type` key attatched:

<img width="876" height="49" alt="Image" src="https://github.com/user-attachments/assets/a5e7a391-378b-4a42-8486-a7fa21c5aedb" />

When reverting back to workgraph 0.7.1, everything works as intended

--- a/src/aiida_workgraph/utils/__init__.py
+++ b/src/aiida_workgraph/utils/__init__.py
@@ -16,6 +16,7 @@ from node_graph.socket import TaggedValue
 from node_graph.socket_spec import SocketSpec
 from aiida.orm.utils.serialize import serialize
 from aiida_workgraph.orm.utils import deserialize_safe
+from copy import deepcopy
 
 
 def inspect_aiida_component_type(executor: Callable) -> str:
@@ -392,7 +393,8 @@ def get_raw_value(identifier, value: Any) -> Any:
         else:
             return value
     elif isinstance(value, orm.Data):
-        content = value.backend_entity.attributes
+        # avoid modifying the original attributes
+        content = deepcopy(value.backend_entity.attributes)
         content['node_type'] = value.node_type
         return content
 
@@ -423,7 +425,6 @@ def process_properties(task: Dict) -> Dict:
 
 def workgraph_to_short_json(wgdata: Dict[str, Union[str, List, Dict]]) -> Dict[str, Union[str, Dict]]:
     """Export a workgraph to a rete js editor data."""
-    from copy import deepcopy
 
     wgdata_short = {
         'name': wgdata['name'],

